api/src/lib/spsp.js

```diff
@@ -59,17 +59,33 @@ module.exports = class SPSP {
   // params should contain:
   // .user.username
   // .destination
-  // .sourceAmount XOR .destinationAmount
+  // .sourceAmount XOR .destinationAmount, float string in currency units
   * quote (params) {
     yield this.factory.connect()
-    return ILP.SPSP.quote(
+
+    // convert float currency units (e.g. '1.23' USD)
+    // to integer amount in the ledger's base units (e.g. '1230000000' nano-USD)
+    const ledgerInfo = yield this.ledger.getInfo()
+    let sourceAmount, destinationAmount
+    if (!isNaN(parseFloat(params.soureAmount))) {
+      sourceAmount = parseFloat(params.sourceAmount) * Math.pow(10, ledgerInfo.scale)
+    }
+    if (!isNaN(parseFloat(params.soureAmount))) {
+      destinationAmount = parseFloat(params.destinationAmount) * Math.pow(10, ledgerInfo.scale)
+    }
+
+    const resultInLedgerUnits = ILP.SPSP.quote(
```

Comment:
The SPSP functions already work in currency units, so this change would be applying the ledger scale twice